### DISCUSSION TITLE
Add Option for Log Start other than midnight.

### DISCUSSION THREAD

**Discussion by stangm** [Chatterino/chatterino2#5386] [Ideas] [answered]
Midnight breaks a lot of streams into two logs, being able to choose a start time would help with this.

**Comment by pajlada** [comment]
Could you share some information about your workflow with logs, what you use them for etc? I think with that information we might think of some other solution that fits better than adding a pure hour offset for log resetting

**Comment by stangm** [comment]
sorry, I missed the reply to this.  A lot of the streamers I follow regularly stream past midnight, so the streams are always broken across two logs.  I want to be able to use the logs to generate stats on subs/bits/follows etc per stream and having one stream per log would be very helpful

**Comment by stangm** [comment]
Alternatively, being able to set a time zone would work as well, so i could share logs with folks in different time zones.  That is difficult today.   I mod for one streamer and another mod in a different time zone has all the logs at different times, if we could set ours to the same time zone that would help.

**Comment by pajlada** [comment, marked as answer]
I've added this PR https://github.com/Chatterino/chatterino2/pull/5507 which adds an option called "Store live stream logs as separate files".
When this is enabled, logs are stored based on their stream ID inside the channel's logging directory.
As an example, I went live/offline on my channel bajlada 3 times, and it wrote the following 3 files to the bajlada logging directory:
 - `bajlada-40851091093.log`
 - `bajlada-40851100165.log`
 - `bajlada-40851104229.log`

all containing messages that took place during those streams.

No date in the filename, since streams can span multiple dates. You can use your computer's file browser sorting/columns to see when the stream took place (or read the top of the log file). Not using stream title or game since those can contain horrendous characters that would not be suitable for filenames, and stream IDs are unique meaning if you close & start up your client while a stream is live, it'll keep writing to the same file.

This has been merged into the main branch and is available in the latest nightly build
